Sources/REventLogger/DataStorage/REventsStorage.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+
+#if SWIFT_PACKAGE
+import class RSDKUtilsMain.AtomicGetSet
+#else
+import class RSDKUtils.AtomicGetSet
+#endif
+
+internal protocol EventDataCacheable {
+
+    func getAllEvents() -> [REvent]
+    func retriveEvent(_ id: String) -> REvent?
+    func getEventCount() -> Int
+    func insertOrUpdateEvent(_ id: String, event: REvent)
+    func deleteEvents(_ ids: [String])
+    func deleteOldEvents(maxCapacity: Int)
+}
+
+internal class EventDataCache: EventDataCacheable {
+
+    private typealias CacheContainer = [String: REvent]
+    private let userDefaults: UserDefaults
+    @AtomicGetSet private var cachedContainer: CacheContainer
+    private let persistedDataKey = "Event_logger_cache"
+
+    init(userDefaults: UserDefaults) {
+        self.userDefaults = userDefaults
+
+        if let persistedData = userDefaults.object(forKey: persistedDataKey) as? Data {
+            do {
+                let decodedData = try JSONDecoder().decode(CacheContainer.self, from: persistedData)
```

Comment:
assign initial value and use `try? JSONDecoder()` then remove `do cache` & `else` part